lib/phoenix_live_view/async.ex

```diff
@@ -50,14 +50,20 @@ defmodule Phoenix.LiveView.Async do
 
     socket
     |> Phoenix.Component.assign(new_assigns)
-    |> run_async_task(keys, wrapped_func, :assign)
+    |> run_async_task(keys, wrapped_func, :assign, opts)
   end
 
-  defp run_async_task(%Socket{} = socket, key, func, kind) do
+  defp run_async_task(%Socket{} = socket, key, func, kind, opts) do
     if Phoenix.LiveView.connected?(socket) do
       lv_pid = self()
       cid = cid(socket)
-      {:ok, pid} = Task.start_link(fn -> do_async(lv_pid, cid, key, func, kind) end)
+      {:ok, pid} = if supervisor = Keyword.get(opts, :supervisor) do
+        {:ok, pid} = Task.Supervisor.start_child(supervisor, fn -> do_async(lv_pid, cid, key, func, kind) end)
+        Process.link(pid)
+        {:ok, pid}
```

Comment:
There is a race condition here in that, if the LV process crashes, the spawned process will continue running. I think it is better to link from the spawned process, as it is less likely to crash as we have control over when it starts running external code:

```suggestion
        Task.Supervisor.start_child(supervisor, fn ->
          Process.link(lv_pid)
          do_async(lv_pid, cid, key, func, kind)
        end)
```